ssc_interface_wrapper_ros2/include/ssc_interface_wrapper/converter_node.hpp

```diff
@@ -0,0 +1,190 @@
+#include <automotive_platform_msgs/msg/speed_mode.hpp>
+#include <automotive_platform_msgs/msg/turn_signal_command.hpp>
+#include <automotive_platform_msgs/msg/gear_command.hpp>
+
+#include <automotive_platform_msgs/msg/gear.hpp>
+
+
+namespace ssc_interface_wrapper{
+
+static const std::string BASE_FRAME_ID = "base_link";
+
+  struct ConverterConfig
+  {
+   
+    bool use_adaptive_gear_ratio_ = true;
+    int command_timeout_ = 1000;
+    double status_pub_rate_ = 30.0;
+    double wheel_base_ = 2.79;
+    double tire_radius_ = 0.39;
+    double ssc_gear_ratio_ = 16.135;
+    double acceleration_limit_ = 3.0;
+    double deceleration_limit_ = 3.0;
+    double max_curvature_rate_ = 0.15;
+    double agr_coef_a_ = 15.713;
+    double agr_coef_b_ = 0.053;
+    double agr_coef_c_ = 0.042;
+
+    // Stream operator for this config
+    
+    friend std::ostream &operator<<(std::ostream &output, const ConverterConfig &c)
+    {
+      output << "ssc_interface_wrapper::ConverterConfig { " << std::endl
+           << "use_adaptive_gear_ratio_: " << c.use_adaptive_gear_ratio_ << std::endl
+           << "command_timeout:" << c.command_timeout_ <<std::endl
+           << "status_pub_rate:" << c.status_pub_rate_<<std::endl
+           << "wheel_base_: " << c.wheel_base_ << std::endl
+           << "tire_radius: " << c.tire_radius_ << std::endl
+           << "ssc_gear_ratio: " << c.ssc_gear_ratio_ << std::endl
+           << "acceleration_limit: " << c.acceleration_limit_ << std::endl
+           << "deceleration_limit: " << c.deceleration_limit_ << std::endl
+           << "max_curvature_rate: "<< c.max_curvature_rate_ <<std::endl
+           << "agr_coef_a_: " << c.agr_coef_a_ << std::endl
+           << "agr_coef_b_: " << c.agr_coef_b_ << std::endl
+           << "agr_coef_c_: " << c.agr_coef_c_ << std::endl
+           << "}" << std::endl;
+      return output;
+    }
+  };
+
+class Converter : public carma_ros2_utils::CarmaLifecycleNode
+{
+    private:
+    
+    carma_ros2_utils::PubPtr<autoware_msgs::msg::VehicleStatus> vehicle_status_pub_; 
+    carma_ros2_utils::PubPtr<geometry_msgs::msg::TwistStamped> current_twist_pub_;
+
+    //autoware.ai publishes a vehicle status topic after subscribing to ssc topics, that needs to be created here
+    //Create vehicle status message after subscribing to ssc topics
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::VelocityAccelCov> velocity_accel_sub_;
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::CurvatureFeedback> curvature_feedback_sub_;
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::ThrottleFeedback> throttle_feedback_sub_;
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::BrakeFeedback> brake_feedback_sub_;
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::GearFeedback> gear_feedback_sub_;
+    carma_ros2_utils::SubPtr<automotive_platform_msgs::msg::SteeringFeedback> steering_wheel_sub_;
```

Comment:
In ROS1 these were synchronized with message_filters https://github.com/usdot-fhwa-stol/autoware.ai/blob/1cb4e74b810111369a6bbe49b0e64e07767454c0/drivers/as/nodes/ssc_interface/ssc_interface.h#L58
It looks like in ROS2 there is a message_filters package available https://github.com/ros2/message_filters/tree/foxy
(This is already in the base image no need to pull in)

If possible preserving the synchronization will likely improve performance.